lib/pbio/include/pbsys/main.h

```diff
@@ -11,9 +11,36 @@
 
 #include <pbsys/config.h>
 
+#include <stdbool.h>
+#include <stdint.h>
+
+/**
+ * Main application program data information.
+ */
+typedef struct _pbsys_main_program_t {
+    /**
+     * User program data.
+     */
+    uint8_t *data;
```

Comment:
```suggestion
    const uint8_t *data;
```

In case data is in ROM.